### PR TITLE
Add jinja2 templates

### PR DESCRIPTION
**Pull request by Ch0live**
## Description

Adds templating for creating complex prompts with insertable variables. Plus a `prompting.py` module containing a class `PromptEngine` to make loading the templates easy. Code changes based on AutoGPT architecture

## Changelog
- Adds `prompting.py` for easy loading of the templates anywhere in the codebase
- Tested using `prompting.py`
- Adds initial templates required for the router
- Removes `version` from docker `compose.yml` (deprecated field)

### CODE REVIEW

File: backend/prompts/prompting.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import logging
+import os
+
+from jinja2 import Environment, FileSystemLoader
+
+
+class PromptEngine:
+    """
+    Class to handle loading and populating Jinja2 templates for prompts.
+    """
```
Comment (bheyman-scottlogic) [inline backend/prompts/prompting.py:10]: We haven't used docstrings anywhere else. Should be consistent across the board. Either by adding them in or removing them here

File: backend/prompts/prompting.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    def __init__(self, model: str):
+        """
+        Initialize the PromptEngine with the specified model.
+
+        Args:
+            model (str): The model to use for loading prompts.
+            debug_enabled (bool): Enable or disable debug logging.
```
Comment (harryBedfordSL) [inline backend/prompts/prompting.py:18]: looks like `debug_enabled` isn't used anywhere?

File: backend/prompts/prompting_test.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    engine = PromptEngine("mistral")
+    try:
+        engine.load_prompt("task-step")
+    except Exception:
+        raise
+
+def test_load_agent_selection_format_template():
+    engine = PromptEngine("mistral")
+    try:
+        engine.load_prompt("agent-selection-format")
```
Comment (harryBedfordSL) [inline backend/prompts/prompting_test.py:19]: is there an assertion we can do here and for the other tests? (Forgive me if I'm missing it)

File: backend/prompts/prompting.py (hunk 2 above)
Comment (Ch0live) [reply backend/prompts/prompting.py:18]: True, I've removed it from the init method signature. Removed

File: backend/prompts/prompting.py (hunk 1 above)
Comment (Ch0live) [reply backend/prompts/prompting.py:10]: I agree, removed also

File: backend/prompts/prompting_test.py (hunk 3 above)
Comment (Ch0live) [reply backend/prompts/prompting_test.py]: I've added full string assertions

**Review by harryBedfordSL — APPROVED** [review]
looks great to me

**Review by bheyman-scottlogic — APPROVED** [review]
Good stuff Charlie!

**Pull request merged**